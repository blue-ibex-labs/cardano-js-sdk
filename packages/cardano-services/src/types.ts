export type ModuleState = null | 'initializing' | 'initialized';
export type Awaited<T> = T extends PromiseLike<infer U> ? U : T;
