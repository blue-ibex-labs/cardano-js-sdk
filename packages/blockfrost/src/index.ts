export { WalletProvider } from '@cardano-sdk/core';
export * from './blockfrostWalletProvider';
export * from './blockfrostAssetProvider';
export * from './blockfrostTxSubmitProvider';
export * from './blockfrostNetworkInfoProvider';
export * from './blockfrostUtxoProvider';
export { Options } from '@blockfrost/blockfrost-js/lib/types';
export { BlockFrostAPI } from '@blockfrost/blockfrost-js';
