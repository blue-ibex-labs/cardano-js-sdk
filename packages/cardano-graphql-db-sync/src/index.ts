export { WalletProvider } from '@cardano-sdk/core';
export * from './cardanoGraphqlDbSyncProvider';
export * from './cardanoGraphqlDbSyncTxSubmitProvider';
