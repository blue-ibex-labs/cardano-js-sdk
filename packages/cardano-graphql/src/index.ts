export { WalletProvider, StakePoolSearchProvider } from '@cardano-sdk/core';
export * from './StakePoolSearchProvider';
export * from './WalletProvider';
export * from './AssetProvider';
export * from './createSDK';
// Do not export this. Using core types elsewhere in the sdk.
// export * as Schema from './Schema';
