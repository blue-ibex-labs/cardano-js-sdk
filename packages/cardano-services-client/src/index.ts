export { TxSubmitProvider } from '@cardano-sdk/core';
export * from './HttpProvider';
export * from './TxSubmitProvider';
export * from './StakePoolSearchProvider';
export * from './UtxoProvider';
