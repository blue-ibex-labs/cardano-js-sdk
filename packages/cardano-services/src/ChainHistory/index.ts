export * from './ChainHistoryHttpService';
export * from './DbSyncChainHistoryProvider';
