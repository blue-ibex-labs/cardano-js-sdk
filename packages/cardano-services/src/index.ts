export * from './Http';
export * from './Program';
export * from './RunnableModule';
export * from './StakePoolSearch';
export * from './TxSubmit';
export * from './Utxo';
export * as errors from './errors';
