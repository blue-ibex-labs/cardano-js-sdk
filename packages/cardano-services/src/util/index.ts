export * from './http';
export * from './logging';
export * from './provider';
export * from './hexString';
