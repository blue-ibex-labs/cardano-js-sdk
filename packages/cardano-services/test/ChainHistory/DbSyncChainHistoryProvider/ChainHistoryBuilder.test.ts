import { Cardano } from '@cardano-sdk/core';
import { ChainHistoryBuilder } from '../../../src';
import { Pool } from 'pg';

describe('ChainHistoryBuilder', () => {
  let dbConnection: Pool;
  let builder: ChainHistoryBuilder;

  beforeAll(async () => {
    dbConnection = new Pool({ connectionString: process.env.DB_CONNECTION_STRING });
    builder = new ChainHistoryBuilder(dbConnection);
  });

  afterAll(async () => {
    await dbConnection.end();
  });

  describe('queryTransactionInputsByHashes', () => {
    test('query transaction inputs by tx hashes', async () => {
      const result = await builder.queryTransactionInputsByHashes([
        Cardano.TransactionId('cefd2fcf657e5e5d6c35975f4e052f427819391b153ebb16ad8aa107ba5a3819'),
        Cardano.TransactionId('952dfa431223fd671c5e9e048e016f70fcebd9e41fcb726969415ff692736eeb')
      ]);
      expect(result).toHaveLength(2);
      expect(result).toMatchSnapshot();
    });
    test('query transaction inputs with empty array', async () => {
      const result = await builder.queryTransactionInputsByHashes([]);
      expect(result).toHaveLength(0);
    });
    test('query transaction inputs when tx hashes not found', async () => {
      const result = await builder.queryTransactionInputsByHashes([
        Cardano.TransactionId('0000000000000000000000000000000000000000000000000000000000000000')
      ]);
      expect(result).toHaveLength(0);
    });
  });

  describe('queryTransactionOutputsByHashes', () => {
    test('query transaction outputs by tx hashes', async () => {
      const result = await builder.queryTransactionOutputsByHashes([
        Cardano.TransactionId('cefd2fcf657e5e5d6c35975f4e052f427819391b153ebb16ad8aa107ba5a3819'),
        Cardano.TransactionId('952dfa431223fd671c5e9e048e016f70fcebd9e41fcb726969415ff692736eeb')
      ]);
      expect(result).toHaveLength(4);
      expect(result).toMatchSnapshot();
    });
    test('query transaction outputs with empty array', async () => {
      const result = await builder.queryTransactionOutputsByHashes([]);
      expect(result).toHaveLength(0);
    });
    test('query transaction outputs when tx hashes not found', async () => {
      const result = await builder.queryTransactionOutputsByHashes([
        Cardano.TransactionId('0000000000000000000000000000000000000000000000000000000000000000')
      ]);
      expect(result).toHaveLength(0);
    });
  });

  describe('queryMultiAssetsByTxOut', () => {
    test('query multi assets by tx output', async () => {
      const result = await builder.queryMultiAssetsByTxOut(['5396853', '5396911', '5396912']);
      expect(result.size).toEqual(3);
      expect(result).toMatchSnapshot();
    });
    test('query multi assets with empty array', async () => {
      const result = await builder.queryMultiAssetsByTxOut([]);
      expect(result.size).toEqual(0);
    });
    test('query multi assets when tx output not found or has no multi-assets', async () => {
      const result = await builder.queryMultiAssetsByTxOut(['0', '5396852']);
      expect(result.size).toEqual(0);
    });
  });

  describe('queryTxMintByHashes', () => {
    test('query transaction mint by tx hashes', async () => {
      const result = await builder.queryTxMintByHashes([
        Cardano.TransactionId('24e75c64a309fd8fb400933795b2522ca818cba80a3838c2ff14cec2cc8ffe4e'),
        Cardano.TransactionId('952dfa431223fd671c5e9e048e016f70fcebd9e41fcb726969415ff692736eeb')
      ]);
      expect(result.size).toEqual(2);
      expect(result).toMatchSnapshot();
    });
    test('query transaction mint with empty array', async () => {
      const result = await builder.queryTxMintByHashes([]);
      expect(result.size).toEqual(0);
    });
    test('query transaction mint when tx not found or has no mint operations', async () => {
      const result = await builder.queryTxMintByHashes([
        Cardano.TransactionId('0000000000000000000000000000000000000000000000000000000000000000'),
        Cardano.TransactionId('cefd2fcf657e5e5d6c35975f4e052f427819391b153ebb16ad8aa107ba5a3819')
      ]);
      expect(result.size).toEqual(0);
    });
  });

  describe('queryWithdrawalsByHashes', () => {
    test('query transaction withdrawals by tx hashes', async () => {
      const result = await builder.queryWithdrawalsByHashes([
        Cardano.TransactionId('cb66e0f5778718f8bfcfd043712f37d9993f4703b254a7a4d954d34225fe2f99')
      ]);
      expect(result.size).toEqual(1);
      expect(result).toMatchSnapshot();
    });
    test('query transaction withdrawals with empty array', async () => {
      const result = await builder.queryWithdrawalsByHashes([]);
      expect(result.size).toEqual(0);
    });
    test('query transaction withdrawals when tx not found or has no withdrawals', async () => {
      const result = await builder.queryWithdrawalsByHashes([
        Cardano.TransactionId('0000000000000000000000000000000000000000000000000000000000000000'),
        Cardano.TransactionId('cefd2fcf657e5e5d6c35975f4e052f427819391b153ebb16ad8aa107ba5a3819')
      ]);
      expect(result.size).toEqual(0);
    });
  });

  describe('queryRedeemersByHashes', () => {
    test('query transaction redeemers by tx hashes', async () => {
      const result = await builder.queryRedeemersByHashes([
        Cardano.TransactionId('5acd6efb1b66299f1c5a2c4221af4bcaa4ba9929e8e6aa0e3f48707fa1796fc3'),
        Cardano.TransactionId('24e75c64a309fd8fb400933795b2522ca818cba80a3838c2ff14cec2cc8ffe4e')
      ]);
      expect(result.size).toEqual(2);
      expect(result).toMatchSnapshot();
    });
    test('query transaction redeemers with empty array', async () => {
      const result = await builder.queryRedeemersByHashes([]);
      expect(result.size).toEqual(0);
    });
    test('query transaction redeemers when tx not found or has no redeemers', async () => {
      const result = await builder.queryRedeemersByHashes([
        Cardano.TransactionId('0000000000000000000000000000000000000000000000000000000000000000'),
        Cardano.TransactionId('cefd2fcf657e5e5d6c35975f4e052f427819391b153ebb16ad8aa107ba5a3819')
      ]);
      expect(result.size).toEqual(0);
    });
  });

  describe('queryTxMetadataByHashes', () => {
    test('query transaction metadata by tx hashes', async () => {
      const result = await builder.queryTxMetadataByHashes([
        Cardano.TransactionId('3d2278e9cef71c79720a11bc3e08acbbd5f2175f7015d358c867fc9b419ae0b2'),
        Cardano.TransactionId('545c4656544054045f5a4db0e962f6b09fc6d98b0303d42f3f006e3d920d3720')
      ]);
      expect(result.size).toEqual(2);
      expect(result).toMatchSnapshot();
    });
    test('query transaction metadata with empty array', async () => {
      const result = await builder.queryTxMetadataByHashes([]);
      expect(result.size).toEqual(0);
    });
    test('query transaction metadata when tx not found or has no metadata', async () => {
      const result = await builder.queryTxMetadataByHashes([
        Cardano.TransactionId('0000000000000000000000000000000000000000000000000000000000000000')
      ]);
      expect(result.size).toEqual(0);
    });
  });

  describe('queryProtocolParams', () => {
    test('query protocol parameters from last epoch', async () => {
      const result = await builder.queryProtocolParams();
      expect(result).toMatchSnapshot();
    });
  });

  describe('queryCertificatesByHashes', () => {
    test('query certificates by tx hashes', async () => {
      const result = await builder.queryCertificatesByHashes([
        Cardano.TransactionId('face165bd7aa8d0d661cf1ceaa4e35d7611be3b1c7997da378c547aa2464a4fd'),
        Cardano.TransactionId('19251f57476d7af2777252270413c01383d9503110a68b4fde1a239c119c4f5d'),
        Cardano.TransactionId('295d5e0f7ee182426eaeda8c9f1c63502c72cdf4afd6e0ee0f209adf94a614e7')
      ]);
      expect(result.size).toEqual(3);
      expect(result).toMatchSnapshot();
    });
    test('query certificates with empty array', async () => {
      const result = await builder.queryCertificatesByHashes([]);
      expect(result.size).toEqual(0);
    });
    test('query certificates when tx not found or has not certificates', async () => {
      const result = await builder.queryCertificatesByHashes([
        Cardano.TransactionId('0000000000000000000000000000000000000000000000000000000000000000'),
        Cardano.TransactionId('cefd2fcf657e5e5d6c35975f4e052f427819391b153ebb16ad8aa107ba5a3819')
      ]);
      expect(result.size).toEqual(0);
    });
  });
});
