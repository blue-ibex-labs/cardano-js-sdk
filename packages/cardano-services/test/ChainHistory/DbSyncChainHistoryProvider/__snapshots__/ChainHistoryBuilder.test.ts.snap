// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChainHistoryBuilder queryCertificatesByHashes query certificates by tx hashes 1`] = `
Map {
  "face165bd7aa8d0d661cf1ceaa4e35d7611be3b1c7997da378c547aa2464a4fd" => Array [
    Object {
      "__typename": "PoolRetirementCertificate",
      "epoch": 179,
      "poolId": "pool1xv0p9vje3r09tac22mhgnsg08rmfm5qxadggjnfnq3cz52apdew",
    },
  ],
  "19251f57476d7af2777252270413c01383d9503110a68b4fde1a239c119c4f5d" => Array [
    Object {
      "__typename": "PoolRegistrationCertificate",
      "poolParameters": null,
    },
    Object {
      "__typename": "StakeDelegationCertificate",
      "poolId": "pool1xv0p9vje3r09tac22mhgnsg08rmfm5qxadggjnfnq3cz52apdew",
      "stakeKeyHash": "34ff9be684ca26d2efee39a693fe051d393bb90d567357e6bbdeb2c8",
    },
  ],
  "295d5e0f7ee182426eaeda8c9f1c63502c72cdf4afd6e0ee0f209adf94a614e7" => Array [
    Object {
      "__typename": "StakeKeyRegistrationCertificate",
      "stakeKeyHash": "c644b3906d4b9952886263832e59a841b5e1b073806747c8d731fbce",
    },
    Object {
      "__typename": "StakeKeyRegistrationCertificate",
      "stakeKeyHash": "8ebe40911f3ae2e6c1fe43dc647566ae1d8701d4d018a1d2a333124b",
    },
    Object {
      "__typename": "PoolRegistrationCertificate",
      "poolParameters": null,
    },
    Object {
      "__typename": "StakeDelegationCertificate",
      "poolId": "pool1y25deq9kldy9y9gfvrpw8zt05zsrfx84zjhugaxrx9ftvwdpua2",
      "stakeKeyHash": "c644b3906d4b9952886263832e59a841b5e1b073806747c8d731fbce",
    },
  ],
}
`;

exports[`ChainHistoryBuilder queryMultiAssetsByTxOut query multi assets by tx output 1`] = `
Map {
  "5396911" => Map {
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf3916522534245525259" => 10000000n,
  },
  "5396853" => Map {
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf391652243484f43" => 10000000n,
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf39165224d494e54" => 5000000000n,
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf3916522524245525259" => 1000000000n,
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf3916522534245525259" => 2000000000n,
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf391652256414e494c" => 15000000n,
  },
  "5396912" => Map {
    "22be35160908d953c25b30d1ad59876cbf073a8b31ef7a4f77468355414141" => 4208n,
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf39165224d494e54" => 1914111968n,
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf3916522524245525259" => 1000000000n,
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf3916522534245525259" => 1990000000n,
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf391652256414e494c" => 15000000n,
    "cd9d85fe5c8a1684c42ae0082e7e912fb7a7cca0900c5a3915eb1c3442454e59" => 2819431n,
    "d311d3488cc4fef19d05634adce8534977a3bc6fc18136ad65df1d4f6c71202e01" => 100000000n,
    "d311d3488cc4fef19d05634adce8534977a3bc6fc18136ad65df1d4f6c712038" => 172696733n,
  },
}
`;

exports[`ChainHistoryBuilder queryProtocolParams query protocol parameters from last epoch 1`] = `
Object {
  "coinsPerUtxoWord": NaN,
  "maxCollateralInputs": 1,
  "maxTxSize": 16384,
  "maxValueSize": 1000,
  "minFeeCoefficient": 44,
  "minFeeConstant": 155381,
  "minPoolCost": 340000000,
  "poolDeposit": 500000000,
  "protocolVersion": Object {
    "major": 6,
    "minor": 0,
  },
  "stakeKeyDeposit": 2000000,
}
`;

exports[`ChainHistoryBuilder queryRedeemersByHashes query transaction redeemers by tx hashes 1`] = `
Map {
  "5acd6efb1b66299f1c5a2c4221af4bcaa4ba9929e8e6aa0e3f48707fa1796fc3" => Array [
    Object {
      "executionUnits": Object {
        "memory": 4837876,
        "steps": 1615750737,
      },
      "index": 0,
      "purpose": "spend",
      "scriptHash": "9f244c64c6ca8e495f4d0ac48f106a1dbdf20b0821d598ebbcd1c57d",
    },
  ],
  "24e75c64a309fd8fb400933795b2522ca818cba80a3838c2ff14cec2cc8ffe4e" => Array [
    Object {
      "executionUnits": Object {
        "memory": 4195651,
        "steps": 1591910324,
      },
      "index": 0,
      "purpose": "spend",
      "scriptHash": "64de10aac4aad00a1a25079498378587c8d92825f17223dcf4fad59c",
    },
    Object {
      "executionUnits": Object {
        "memory": 88558,
        "steps": 46839629,
      },
      "index": 1,
      "purpose": "spend",
      "scriptHash": "6e4664e3f6acc48e849eacf4e3607ee6ae2758d6d8b80fa9d8fdd761",
    },
    Object {
      "executionUnits": Object {
        "memory": 88558,
        "steps": 46839629,
      },
      "index": 3,
      "purpose": "spend",
      "scriptHash": "6e4664e3f6acc48e849eacf4e3607ee6ae2758d6d8b80fa9d8fdd761",
    },
    Object {
      "executionUnits": Object {
        "memory": 88558,
        "steps": 46839629,
      },
      "index": 5,
      "purpose": "spend",
      "scriptHash": "6e4664e3f6acc48e849eacf4e3607ee6ae2758d6d8b80fa9d8fdd761",
    },
    Object {
      "executionUnits": Object {
        "memory": 106622,
        "steps": 55720739,
      },
      "index": 0,
      "purpose": "mint",
      "scriptHash": "22c3b86a5b88a78b5de52f4aed2831d1483b3b7681f1ee2569538130",
    },
  ],
}
`;

exports[`ChainHistoryBuilder queryTransactionInputsByHashes query transaction inputs by tx hashes 1`] = `
Array [
  Object {
    "address": "addr_test1qr74u5d3d07x53pq37acghulu7wvjt794s53xp2fz343drhx2aghqc4747n2yhu0dht5xz0t5j6n0fju3rs8mfj83tzqly49jr",
    "index": 0,
    "txId": "952dfa431223fd671c5e9e048e016f70fcebd9e41fcb726969415ff692736eeb",
  },
  Object {
    "address": "addr_test1qpvl4gf5r5h5w4yjrf58rt8slltem9tqfge36zqhsz5kgc8f2tc4ydn4zz9p0eymajqa4vj87he7yz6ssmh9x9h685hst9rgvk",
    "index": 0,
    "txId": "cefd2fcf657e5e5d6c35975f4e052f427819391b153ebb16ad8aa107ba5a3819",
  },
]
`;

exports[`ChainHistoryBuilder queryTransactionOutputsByHashes query transaction outputs by tx hashes 1`] = `
Array [
  Object {
    "address": "addr_test1qr74u5d3d07x53pq37acghulu7wvjt794s53xp2fz343drhx2aghqc4747n2yhu0dht5xz0t5j6n0fju3rs8mfj83tzqly49jr",
    "datum": undefined,
    "index": 0,
    "txId": "952dfa431223fd671c5e9e048e016f70fcebd9e41fcb726969415ff692736eeb",
    "value": Object {
      "assets": undefined,
      "coins": 5528822746696n,
    },
  },
  Object {
    "address": "addr_test1qzmpmxyl7prdf0pazt0futtrg6k9drcay3ednatxqm5ky854nqdumcu3xny9qu5sk9m0wm0ypyty84kxj0gwd9seppfqzs9z2n",
    "datum": undefined,
    "index": 1,
    "txId": "952dfa431223fd671c5e9e048e016f70fcebd9e41fcb726969415ff692736eeb",
    "value": Object {
      "assets": Map {
        "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf391652243484f43" => 10000000n,
        "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf39165224d494e54" => 5000000000n,
        "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf3916522524245525259" => 1000000000n,
        "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf3916522534245525259" => 2000000000n,
        "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf391652256414e494c" => 15000000n,
      },
      "coins": 1000000000n,
    },
  },
  Object {
    "address": "addr_test1wrsexavz37208qda7mwwu4k7hcpg26cz0ce86f5e9kul3hqzlh22t",
    "datum": "c5dfa8c3cbd5a959829618a7b46e163078cb3f1b39f152514d0c3686d553529a",
    "index": 0,
    "txId": "cefd2fcf657e5e5d6c35975f4e052f427819391b153ebb16ad8aa107ba5a3819",
    "value": Object {
      "assets": Map {
        "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf3916522534245525259" => 10000000n,
      },
      "coins": 7420514n,
    },
  },
  Object {
    "address": "addr_test1qpvl4gf5r5h5w4yjrf58rt8slltem9tqfge36zqhsz5kgc8f2tc4ydn4zz9p0eymajqa4vj87he7yz6ssmh9x9h685hst9rgvk",
    "datum": undefined,
    "index": 1,
    "txId": "cefd2fcf657e5e5d6c35975f4e052f427819391b153ebb16ad8aa107ba5a3819",
    "value": Object {
      "assets": Map {
        "22be35160908d953c25b30d1ad59876cbf073a8b31ef7a4f77468355414141" => 4208n,
        "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf39165224d494e54" => 1914111968n,
        "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf3916522524245525259" => 1000000000n,
        "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf3916522534245525259" => 1990000000n,
        "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf391652256414e494c" => 15000000n,
        "cd9d85fe5c8a1684c42ae0082e7e912fb7a7cca0900c5a3915eb1c3442454e59" => 2819431n,
        "d311d3488cc4fef19d05634adce8534977a3bc6fc18136ad65df1d4f6c71202e01" => 100000000n,
        "d311d3488cc4fef19d05634adce8534977a3bc6fc18136ad65df1d4f6c712038" => 172696733n,
      },
      "coins": 850065083n,
    },
  },
]
`;

exports[`ChainHistoryBuilder queryTxMetadataByHashes query transaction metadata by tx hashes 1`] = `
Map {
  "3d2278e9cef71c79720a11bc3e08acbbd5f2175f7015d358c867fc9b419ae0b2" => Map {
    1968n => Map {
      "LQADA" => Array [
        Map {
          "value" => "0.00006725",
          "source" => "muesliswap",
        },
      ],
      "ADABTC" => Array [
        Map {
          "value" => "0.00002789",
          "source" => "coinGecko",
        },
      ],
      "ADAEUR" => Array [
        Map {
          "value" => "0.934303",
          "source" => "coinGecko",
        },
      ],
      "ADAJPY" => Array [
        Map {
          "value" => "119.97",
          "source" => "coinGecko",
        },
      ],
      "ADAUSD" => Array [
        Map {
          "value" => "1.04",
          "source" => "coinGecko",
        },
      ],
      "WMTADA" => Array [
        Map {
          "value" => "7.95e-7",
          "source" => "muesliswap",
        },
      ],
      "BTCDIFF" => Array [
        Map {
          "value" => "26643185256535.46",
          "source" => "trezor",
        },
      ],
      "HoskyADA" => Array [
        Map {
          "value" => "5.9e-8",
          "source" => "muesliswap",
        },
      ],
      "SpacecoinsADA" => Array [
        Map {
          "value" => "0.00955",
          "source" => "muesliswap",
        },
      ],
    },
  },
  "545c4656544054045f5a4db0e962f6b09fc6d98b0303d42f3f006e3d920d3720" => Map {
    1968n => Map {
      "ADABTC" => Array [
        Map {
          "value" => "0.00002692",
          "source" => "coinGecko",
        },
      ],
      "ADAEUR" => Array [
        Map {
          "value" => "1.18",
          "source" => "coinGecko",
        },
      ],
      "ADAJPY" => Array [
        Map {
          "value" => "151.53",
          "source" => "coinGecko",
        },
      ],
      "ADAUSD" => Array [
        Map {
          "value" => "1.33",
          "source" => "coinGecko",
        },
      ],
      "BTCDIFF" => Array [
        Map {
          "value" => "24195286980613.62",
          "source" => "trezor",
        },
      ],
    },
  },
}
`;

exports[`ChainHistoryBuilder queryTxMintByHashes query transaction mint by tx hashes 1`] = `
Map {
  "24e75c64a309fd8fb400933795b2522ca818cba80a3838c2ff14cec2cc8ffe4e" => Map {
    "22c3b86a5b88a78b5de52f4aed2831d1483b3b7681f1ee2569538130f32324c10a93fe1962cf7470543932cc064482c3495fe3de7502f9edd84ca114" => 15355454n,
  },
  "952dfa431223fd671c5e9e048e016f70fcebd9e41fcb726969415ff692736eeb" => Map {
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf391652243484f43" => 10000000n,
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf39165224d494e54" => 5000000000n,
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf3916522524245525259" => 1000000000n,
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf3916522534245525259" => 2000000000n,
    "57fca08abbaddee36da742a839f7d83a7e1d2419f1507fcbf391652256414e494c" => 15000000n,
  },
}
`;

exports[`ChainHistoryBuilder queryWithdrawalsByHashes query transaction withdrawals by tx hashes 1`] = `
Map {
  "cb66e0f5778718f8bfcfd043712f37d9993f4703b254a7a4d954d34225fe2f99" => Array [
    Object {
      "quantity": 229082n,
      "stakeAddress": "stake_test1urpklgzqsh9yqz8pkyuxcw9dlszpe5flnxjtl55epla6ftqktdyfz",
    },
    Object {
      "quantity": 1409612n,
      "stakeAddress": "stake_test1uqvszkuv7gwzcfshrm37ywcxuqhh7mqx95ey3cm3cx8s76c7w97f8",
    },
  ],
}
`;
